File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    arr = triangle
    for i in range(1, len(triangle)):
        for j in range(len(triangle[i])):
            if j == 0:
                arr[i][j] += triangle[i-1][j]
            elif j == len(triangle[i])-1:
                arr[i][j] += triangle[i-1][j-1]
            else:
                if arr[i-1][j-1] > arr[i-1][j]:
                    arr[i][j] += arr[i-1][j-1]
                else:
                    arr[i][j] += arr[i-1][j]
        
    return max(arr[-1])

File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        int[][] triangle = new int[N + 1][N + 1];
        for (int i = 1; i <= N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            for (int j = 1; j <= i; j++) {
                triangle[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        for (int i = 2; i <= N; i++) {
            for (int j = 1; j <= i; j++) {
                triangle[i][j] += Math.max(triangle[i - 1][j - 1], triangle[i - 1][j]);
            }
        }

        int answer = 0;
        for (int i = 1; i <= N; i++) {
            if (answer < triangle[N][i]) answer = triangle[N][i];
        }

        System.out.println(answer);
    }
}